project/SimpleHTTPServer.scala

```diff
@@ -20,9 +20,19 @@ class SimpleHTTPServer(webDirectory: File, port: Int) extends Closeable {
   // needed for the future flatMap/onComplete in the end
   private implicit val executionContext = system.dispatcher
 
-  private val route = pathPrefix("") {
-    getFromDirectory(webDirectory.getAbsolutePath) ~ getFromFile(new File(webDirectory, "index.html"))
-  }
+  private val route =
+    getFromDirectory(webDirectory.getAbsolutePath) ~
+      // TODO: this needs improving. IT only supports one-folder deep fallback. It'd be good to have a generic fallback
+      // to serving 'index.html' from the appropriate folder anytime the path in the request leaves the last
+      // segment (filename) blank.
+      pathPrefix(Segment) { folderName =>
```

Comment:
Would `RestPath` instead of `Segment` solve this? http://doc.akka.io/api/akka-stream-and-http-experimental/2.0.4/#akka.http.scaladsl.server.PathMatchers$RestPath$